.eslintrc.js

```diff
@@ -1,40 +1,100 @@
 module.exports = {
-  env: {
-    browser: true,
-    es2021: true,
-    jest: true,
-  },
+  // Configuration for JavaScript files
   extends: [
-    'plugin:react/recommended',
-    'airbnb',
-    'prettier',
-    'prettier/react',
-    'plugin:react-hooks/recommended',
-    'plugin:storybook/recommended',
+    "airbnb-base",
+    "next/core-web-vitals", // Needed to avoid warning in next.js build: 'The Next.js plugin was not detected in your ESLint configuration'
+    "plugin:prettier/recommended",
   ],
-  globals: {
-    React: 'writable',
-  },
-  parser: 'babel-eslint',
-  parserOptions: {
-    ecmaFeatures: {
-      jsx: true,
-    },
-    ecmaVersion: 12,
-    sourceType: 'module',
-  },
-  plugins: ['react', 'prettier'],
   rules: {
-    'prettier/prettier': 'error',
-    'react/jsx-props-no-spreading': 'off',
-    'react/destructuring-assignment': 'off',
-    'import/prefer-default-export': 'off',
-    'react/react-in-jsx-scope': 'off',
-    'react/jsx-filename-extension': [
-      'warn',
+    "prettier/prettier": [
+      "error",
       {
-        extensions: ['.jsx', '.js'],
+        singleQuote: false,
+        endOfLine: "auto",
       },
     ],
   },
+  overrides: [
+    // Configuration for TypeScript files
+    {
+      files: ["**/*.ts", "**/*.tsx"],
+      plugins: [
+        "@typescript-eslint",
+        "unused-imports",
+        "tailwindcss",
+        "simple-import-sort",
+        "jsx-a11y",
+      ],
+      extends: [
+        "plugin:tailwindcss/recommended",
+        "airbnb",
+        "airbnb-typescript",
+        "airbnb/hooks",
+        "next/core-web-vitals",
+        "plugin:prettier/recommended",
+        "plugin:jsx-a11y/recommended",
+      ],
+      parserOptions: {
+        project: "./tsconfig.json",
+      },
+      rules: {
+        "prettier/prettier": [
+          "error",
+          {
+            singleQuote: true,
+            endOfLine: "auto",
+          },
+        ],
+        "react/function-component-definition": "off", // Disable Airbnb's specific function type
+        "react/destructuring-assignment": "off", // Vscode doesn't support automatically destructuring, it's a pain to add a new variable
+        "react/require-default-props": "off", // Allow non-defined react props as undefined
+        "react/jsx-props-no-spreading": "off", // _app.tsx uses spread operator and also, react-hook-form
+        "react-hooks/exhaustive-deps": "off", // Incorrectly report needed dependency with Next.js router
+        "@next/next/no-img-element": "off", // We currently not using next/image because it isn't supported with SSG mode
+        "@typescript-eslint/comma-dangle": "off", // Avoid conflict rule between Eslint and Prettier
```

Comment:
Se usar o plugin de eslint to prettier ele automaticamente já remove os conflitos e não vamos precisar dessa regra